pandas/core/groupby/generic.py

```diff
@@ -1731,7 +1731,11 @@ def _wrap_aggregated_output(
         DataFrame
         """
         indexed_output = {key.position: val for key, val in output.items()}
-        columns = Index(key.label for key in output)
+        if self.axis == 0:
+            name = self._obj_with_exclusions.columns.name
+        else:
+            name = self._obj_with_exclusions.index.name
+        columns = Index([key.label for key in output], name=name)
```

Comment:
Does this solve any open issues with the name?

Reply:
Not that I can find. I searched for issues tagged with groupby and for the text "multiindex", "quantile", "any", "all", or "shift", and none were relevant.

Reply:
can we have a test which exercises this?

Reply:
This is hit by, and was added because of, test_multilevel.test_frame_group_ops. The test does not flag other aggregation functions that went through this path because they aren't in the AGG_FUNCTIONS list within this file. Attempting to add e.g. quantile to the list doesn't immediately work because the tests assume that the agg function accepts skipna.